qiskit/transpiler/passes/synthesis/high_level_synthesis.py

```diff
@@ -220,16 +227,34 @@ class HLSConfig:
     :ref:`using-high-level-synthesis-plugins`.
     """
 
-    def __init__(self, use_default_on_unspecified=True, **kwargs):
+    def __init__(
+        self,
+        use_default_on_unspecified=True,
+        plugin_selection: str = "sequential",
+        plugin_evaluation_fn: Optional[Callable[[QuantumCircuit], int]] = None,
```

Comment:
As mentioned in the PR description, https://github.com/Qiskit/qiskit/pull/12203 adds support for HLS plugins to return a `DAGCircuit` (with a `use_dag` flag), and that PR internally sets `use_dag` to `True` in the `HighLevelSynthesis` transpiler pass. Looking at this PR and #12107, I think that my initial approach might lack some flexibility, because it fixes the output of `HighLevelSynthesis` to a `DAGCircuit`. I see a few things we could do:

1. If we follow https://github.com/Qiskit/qiskit/pull/12203, then we'll fix the output of `HighLevelSynthesis` to always be a `DAGCircuit` (which now I start to see that is probably not the intended use). We can then adapt the `plugin_evaluation_fn` definition accordingly. 
2. I can add a `use_dag` argument to `HighLevelSynthesis` too and maybe make it default to `True`? This way we could have it as the default behavior but also support `QuantumCircuit` (or `Operator` in the future if this is useful).

I think that the question boils down to, are we ok with `HighLevelSynthesis` always returning a `DAGCircuit` like other transpiler passes? or do we want to also cater to standalone use? This decision will also affect #12107.

Reply:
In the interest of time, I think I can build #12203 on top of this PR and we can move the discussion there.